504.xml

```diff
@@ -0,0 +1,404 @@
+0: Pull only
+1: Push only
+2: Both. In this case the LwM2M Server MAY choose the preferred mechanism for conveying the profile image to the LwM2M Client.
+3: RSP management performed outside of LwM2M (e.g. change of SIM card, SMS triggered management).
+In case of options 0, 1, or 2 the LwM2M client provides, in addition to resource 3 or 4, at least resources 5, 6, and 7.]]></Description>
+			</Item>
+			<Item ID="12">
+				<Name>Free Memory on SIM</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>Integer</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units>bit</Units>
+				<Description><![CDATA[Estimated current available amount of storage space on SIM which can store data and software in the LwM2M Device (expressed in kilobytes).]]></Description>
+			</Item>
+			<Item ID="13">
+				<Name>Total Memory on SIM</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>Integer</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units>bit</Units>
+				<Description><![CDATA[Total amount of storage space which can store data and software in the LwM2M Device]]></Description>
+			</Item>
+			<Item ID="14">
+				<Name>Integrated Circuit Card Identifier (ICCID)</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Multiple</MultipleInstances>
+				<Mandatory>Mandatory</Mandatory>
+				<Type>String</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description><![CDATA[This resource provides the unique identification number for each installed profile of the current SIM in case of RSP support or
+the unique identification number of the current SIM in case of current SIM being a UICC with no RSP support. Please refer ETSI TS 102.22.1. In case of multiple installed profiles,
+the currently active profile is the first instance.]]></Description>
+			</Item>
+			<Item ID="15">
+				<Name>eUICC ID</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>String</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description><![CDATA[eUICC-ID (a.k.a. EID), see GSMA SGP.02 and GSMA SGP.29 for definitions]]></Description>
+			</Item>
+			<Item ID="16">
+				<Name>Profile Type</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>Integer</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description><![CDATA[Operator specific defined type of Profile. This is equivalent to the "Profile Description ID" as described in Annex B of GSMA SGP.21 [4]]]></Description>
+			</Item>
+			<Item ID="17">
```

Comment:
Abhi will provide feedback in a few days 4/13/2021

Reply:
Removed the extraneous "[4]" from the description as it was erroneously copied from SGP.21.

To answer how the Profile Type is used:
According to SCG.22, “Profile Type” is equivalent to “Profile Description ID” of SGP.21 Annex B.
According to SGP.21 Annex B, “Profile Description ID” defines as follows: The Operator defines its different profile types (identified by a [non-standardised] Profile Description ID) which contains the Network Access Application like USIM, file structure, data and applications, etc.
As per SGP.22: “The Operator can provide the ICCID and/or the Profile Type. In case where the Profile Type is provided, the SM-DP+ is free to select one of the Profiles that matches the Profile Type.”

Reply:
Thanks Alan! The part that MNO can present it to the SM-DP+ over ES2+ I/F instead of the ICCID is clear. What is not clear is how this information is provided to the device/eUICC such that a LwM2M server can request this information from the LwM2M client. This is not something specified in SGP.22 at least, but in TrustedConnectivityAlliance specification "eUICC Profile Package: Interoperable Format Technical Specification" there is an OPTIONAL field in ProfileHeader called "profileType" with the description "The "profileType" is a free optional text indicating for example, the name of the Profile issuer and the type of Profile.". As far as I can understand this string is not necessarily the same as the Profile Type in SGP.22. If it is the profileType field of ProfileHeader that is to be returned by this resource then it is better to refer to TCA spec rather then SGP.22, I think. Given an GSMA compliant eUICC I am not sure if/how this information is stored after profile decryption and installation and how you can obtain this information. I believe also nuSIM is reusing ProfileHeader structure from TCA spec, and maybe in nuSIM case it is possible to read out this field from nuSIM.

Reply:
Changed type to "String"

Reply:
Updated the description as suggested in object 505.